src/main/java/com/artipie/asto/etcd/EtcdStorage.java

```diff
@@ -108,14 +127,22 @@ public EtcdStorage(final Client client) {
 
     @Override
     public CompletableFuture<? extends Meta> metadata(final Key key) {
-        return CompletableFuture.completedFuture(Meta.EMPTY);
+        return this.client.getKVClient().get(keyToSeq(key)).thenApply(
+            rsp -> rsp.getKvs().stream().max(
```

Comment:
@baudoliver7 is it possible to fetch metadata without loading full content of key-values?

Reply:
@g4s8 We can do that, but we will not have size of value. We can have other metadata. And unfortunately, there is no way to get size of value without the value itself. I checked in [etcd developer guide](https://etcd.io/docs/v3.5/dev-guide/interacting_v3/) and there is nothing there that mention that.